src/components/calendar/bl-calendar.ts

```diff
@@ -0,0 +1,133 @@
+import { CSSResultGroup, html, LitElement } from "lit";
+import { customElement, property, state } from "lit/decorators.js";
+import "../button/bl-button";
+import "../icon/bl-icon";
+import { DAYS, MONTHS } from "./bl-calendar.constant";
+import style from "./bl-calendar.css";
+import { TCalendarTypes, TMonth } from "./bl-calendar.types";
+
+@customElement("bl-calendar")
+export default class BlCalendar extends LitElement {
+  @property()
+  type: TCalendarTypes = "single";
+
+  @property()
+  minDate?: Date | string;
+
+  @property()
+  maxDate?: Date | string;
+
+  @property({ type: Array, attribute: "default-value", reflect: true })
+  defaultValue?: Date | Date[];
+
+  @property({ type: Number, attribute: "start-of-week", reflect: true })
+  startOfWeek?: number = 1;
+
+  @property()
+  locale?: string = document.documentElement.lang;
+
+  @property({ type: Array })
+  disabledDates?: Date | Date[];
+
+  @state()
+  private day: number;
+
+  @state()
+  private month: TMonth = MONTHS[new Date().getMonth()];
+
+  @state()
+  private year: number = new Date().getFullYear();
+
+  static get styles(): CSSResultGroup {
+    return [style];
+  }
+  render() {
```

Comment:
```suggestion
  render(): TemplateResult {
```

Reply:
Additionally, moving the render method to the bottom of the code would increase readability